src/lib.rs

```diff
@@ -236,3 +256,42 @@ impl<H> CuckooFilter<H>
         }
     }
 }
+
+/// A minimal representation of the CuckooFilter which can be transfered or stored, then recovered at a later stage.
```

Comment:
It would be nice to explain more of the reasonings behind the separate `ExportedCuckooFilter`s so we don't forget it: That `value` is a serialized version of the CuckooFilter's memory, each bucket and each fingerprint chained after another, and that the `length` field is not `values.length()`, but the number of valid fingerprints inside the CuckooFilter, to save the time of going through all fingerprints and checking how many are not equal to the null pattern.

I'm wondering whether it would make sense to add `FINGERPRINT_SIZE` and `BUCKET_SIZE` values to that struct and check whether the CuckooFilter was stored with the same values, or gets loaded from a version where those constants were changed and compile time, to prevent breakage / make it future-proof for future versions of rust-cuckoofilter where a user can choose those constants. @seiflotfy what do you think?

Reply:
Sure, I can write that up. Will wait until we confirm if the constants should be added or not.

Reply:
I've added some documentation here, but haven't added the constants yet.

I guess it's debatable weather there's a use case for altering these values at all based on the discussion in the paper. However, if it is something that changes in a future version, that wouldn't be hard to add later here too. We can serde to use defaults if values are not found for example.

Reply:
Nicely documented. :+1:

Regarding the parameters, I'd say those really are application dependent, e.g. when you want to put in large amounts of items, and / or are low on memory, and / or can tolerate a higher or require a lower false positive rate than the default parameters allow.

But as the default parameters with “up to a few million items, using a few megabytes of RAM, having up to a few percent false positives” covers many applications and we don't provide an interface for changing those except editing the source code, we're probably fine with adding those later on whenever we have an interface for changing those, like @Libbum suggested using default values to stay compatible with old serializations.

Whoever changes the parameters by hand needs to know what they're doing. ;)